### PR TITLE
Clear devNetParams and mimic behavior of other param types

### PR DESCRIPTION
**Pull request by codablock**
Mainnet, testnet and regtest params are global and thus zero initialized,
we should mimic the same for devnet params.

Not doing so results in all kinds of issues on devnet.

### CODE REVIEW

File: src/chainparamsbase.cpp [hunk 1]
```diff
@@ -9,6 +9,7 @@
 #include "util.h"
 
 #include <assert.h>
+#include <string.h>
```
Comment (UdjinM6) [inline src/chainparamsbase.cpp]: this looks unrelated...

Comment (UdjinM6) [reply src/chainparamsbase.cpp]: Weird.. `string.h` is included at least in `util.h`, so should be compiling just fine imo.

Comment (codablock) [reply src/chainparamsbase.cpp]: Ok I'll remove that include then.
